lib/CodeGen/CGDynamicCheck.cpp

```diff
@@ -111,6 +111,66 @@ void CodeGenFunction::EmitDynamicBoundsCheck(const Address PtrAddr, const Bounds
   EmitDynamicCheckBlocks(Builder.CreateAnd(LowerChk, UpperChk, "_Dynamic_check.range"));
 }
 
+void CodeGenFunction::EmitDynamicBoundsCheck(const BoundsExpr *DynamicBounds,
+                                             const BoundsExpr *SrcBounds) {
+  if (!getLangOpts().CheckedC)
+    return;
+
+  if (!SrcBounds || !DynamicBounds)
+    return;
+
+  if (SrcBounds->isAny() || SrcBounds->isInvalid())
+    return;
+
+  if (DynamicBounds->isAny() || DynamicBounds->isInvalid())
```

Comment:
You can't write a bounds of `Any` in the language, so I think we should assert if we see a DynamicBounds of any.